### PR TITLE
Update Selenium to v4 + .NET 6 Tests

### PR DESCRIPTION
**Pull request by slang25**
This PR contains 3 changes:
- Updating the `Selenium.WebDriver` dependency to `4.0.1`
- Changes test projects to .NET 6
- Get the test passing on macOS, this includes changing the self-hosted server to aspnetcore

### CODE REVIEW

**Comment by petersladek** [comment]
@obstar Any plans when this will be merged?

**Comment by mhlavacek** [comment]
Any updates on this?  I'm waiting to be able to update to selenium webdriver 4.x

**Comment by slang25** [comment]
That'll be a question for @obstar, would be good to know if there is an intention to merge. No pressure here, just nice to have from my perspective

**Comment by sbowler** [comment]
I'm pretty sure there's a runtime issue taking screenshots that needs to be adjusted with this move to .NET 6. See https://github.com/SeleniumHQ/selenium/issues/8760 for more details on what I'm running into and possible solutions. I could be wrong, but I think this should be adjusted here.

**Comment by ben-soler** [comment]
Any updates on this PR?

**Comment by DDSGooch** [comment]
When will this be merged? Is there any reason this can't be done now. We'd really appreciate this update and I'm happy to do some testing on this. Thanks

**Comment by adiel** [comment]
Hi! Just checking back in on this project and seems like nobody maintaining it any more.

Any volunteers from this thread? I know a lot of projects are relying on it still.

**Review by sbowler — CHANGES_REQUESTED** [review]
I would also suggest updating the NUnit project to net6.0 instead of net45. At this point it would probably be good to update some of the other dependencies as well like browser drivers for example.

File: src/Coypu.Drivers.Tests/DriverSpecs.cs [hunk 1]
```diff
@@ -8,18 +8,19 @@
...
 using NUnit.Framework;
+using OpenQA.Selenium.Chrome;
 using ElementFinder = Coypu.Finders.ElementFinder;
 using FrameFinder = Coypu.Finders.FrameFinder;
 
 [SetUpFixture]
 public class AssemblyTearDown
 {
-    public static SelfishSite TestSite;
+    public static SelfHostedSite TestSite;
```
[the remark this answers is not in the document]
Comment (sbowler) [reply src/Coypu.Drivers.Tests/DriverSpecs.cs:18]: I was using visual studio 2022

**Comment by sbowler** [comment]
@slang25 I think I've highlighted the main issues I found. Just wondering if you'll be able to update this and an idea on the timeframe?

**Comment by slang25** [comment]
I'll have a look over the weekend, you are right, it's totally broken, I blame past me 😆

**Comment by obstar** [comment]
@ijabit tbh the best alternative is playwright a lot faster and user friendly than selenium 🤷‍♂️

**Comment by ijabit** [comment]
> 

Thanks! I think I'm locked in to Selenium though due to organizational choices

**Comment by adiel** [comment]
For sure.

Although a Coypu driver for Playwright should be pretty trivial and best of
both worlds if you ask me. It still has all the semantic finder and wait
issues that this project tackles.

I'm planning to do this in a few weeks time when I have some time off
coming up.

**Comment by ijabit** [comment]
I thought Coypu was tied to Selenium, interesting that you might be able to hook it up to Playwright

**Comment by sbowler** [comment]
@slang25 If you don't have time to work on this, let me know and I can finish it up and submit a PR if you need.

**Comment by slang25** [comment]
I'm not sure I'm going to get around to it 😔

**Comment by adiel** [comment]
I did get this branch working the other day - it was basically just the
errors that were pointed out on the pull request. Then I had to update a
bunch of references and language settings - I didn't really know what I was
doing not having run dotnet for so many years. Managed to get it compiling
though on my Mac, and installed Playwright ok and made the first steps
towards a Playwright driver. Hoping to get time to finish it in a couple of
weeks time.

I'll push what I did (minus the playwright stuff) later when I get in and
maybe you can give it a run and see if I did anything stupid Stacy?

**Comment by ijabit** [comment]
Any updates on this one?

**Comment by sbowler** [comment]
Closing in favor of https://github.com/featurist/coypu/pull/228

**Comment by adiel** [comment]
> I thought Coypu was tied to Selenium, interesting that you might be able to hook it up to Playwright

https://github.com/featurist/coypu/pull/231

**Pull request closed without merging**